modules/api/src/main/java/org/apache/ignite/sql/Session.java

```diff
@@ -98,7 +145,42 @@ default ResultSet execute(@Nullable Transaction transaction, Statement statement
      * @return Operation future.
      * @throws SqlException If failed.
      */
-    CompletableFuture<AsyncResultSet> executeAsync(@Nullable Transaction transaction, Statement statement, @Nullable Object... arguments);
+    CompletableFuture<AsyncResultSet<SqlRow>> executeAsync(
+            @Nullable Transaction transaction,
+            Statement statement,
+            @Nullable Object... arguments);
+
+    /**
+     * Executes SQL statement in an asynchronous way and maps results to objects with the provided mapper.
+     *
+     * @param transaction Transaction to execute the statement within or {@code null}.
+     * @param mapper Mapper.
+     * @param query SQL query template.
+     * @param arguments Arguments for the statement.
+     * @return Operation future.
+     * @throws SqlException If failed.
```

Comment:
this is probably copy-pasted javadoc but anyways... async method won't throw any exception, but complete the future exceptionally

Reply:
Removed